grc/core/params/param.py

```diff
@@ -178,6 +178,10 @@ def evaluate(self):
         # ID and Enum types (not evaled)
         #########################
         if dtype in ('id', 'stream_id') or self.is_enum():
+            if self.options.attributes:
+                expr = collections.UserString(expr)
+                for key, value in self.options.attributes[expr].items():
+                    setattr(expr, key, value)
```

Comment:
to make this save, we should probably prevent str methods from being overwritten here